src/main/java/htsjdk/samtools/CRAMContainerStreamWriter.java

```diff
@@ -317,18 +317,14 @@ protected void flushContainer() throws IllegalArgumentException, IllegalAccessEx
         containerFactory.setPreserveReadNames(preserveReadNames);
 
         int index = 0;
-        int prevAlStart = start;
         for (final SAMRecord samRecord : samRecords) {
             if (samRecord.getReferenceIndex() != SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX && refSeqIndex != samRecord.getReferenceIndex()) {
                 // this may load all ref sequences into memory:
                 sam2CramRecordFactory.setRefBases(source.getReferenceBases(samFileHeader.getSequence(samRecord.getReferenceIndex()), true));
             }
             final CramCompressionRecord cramRecord = sam2CramRecordFactory.createCramRecord(samRecord);
             cramRecord.index = ++index;
-            cramRecord.alignmentDelta = samRecord.getAlignmentStart() - prevAlStart;
```

Comment:
Sometimes this was used (delta encoding case) and sometimes it was not.  We know how to calculate delta on read/write, so do it then, and only if necessary.